tools/gulp/tasks/components.ts

```diff
@@ -1,139 +1,150 @@
 import {task, watch, src, dest} from 'gulp';
+import {ScriptTarget, ModuleKind} from 'typescript';
 import * as path from 'path';
 
 import {
   DIST_COMPONENTS_ROOT, PROJECT_ROOT, COMPONENTS_DIR, HTML_MINIFIER_OPTIONS, LICENSE_BANNER
 } from '../constants';
 import {
-  sassBuildTask, tsBuildTask, execNodeTask, copyTask, sequenceTask,
+  sassBuildTask, tsBuildTask, execNodeTask, sequenceTask,
   triggerLivereload
 } from '../util/task_helpers';
 
-// No typings for these.
+// There are no type definitions available for these imports.
 const inlineResources = require('../../../scripts/release/inline-resources');
 const gulpRollup = require('gulp-better-rollup');
-const gulpMinifyCss = require('gulp-clean-css');
 const gulpMinifyHtml = require('gulp-htmlmin');
 const gulpIf = require('gulp-if');
 
+/** Path to tsconfig file for the components. */
+const tsconfigPath = path.join(COMPONENTS_DIR, 'tsconfig.json');
 
-// NOTE: there are two build "modes" in this file, based on which tsconfig is used.
-// When `tsconfig.json` is used, we are outputting ES6 modules and a UMD bundle. This is used
-// for serving and for release.
-//
-// When `tsconfig-spec.json` is used, we are outputting CommonJS modules. This is used
-// for unit tests (karma).
-
-/** Path to the tsconfig used for ESM output. */
-const tsconfigPath = path.relative(PROJECT_ROOT, path.join(COMPONENTS_DIR, 'tsconfig-srcs.json'));
+/** Asset files to be added to the components output. */
+const assetFiles = [
+  path.join(COMPONENTS_DIR, '**/*.html'),
+  path.join(COMPONENTS_DIR, 'package.json'),
+  path.join(PROJECT_ROOT, 'README.md'),
+  path.join(PROJECT_ROOT, 'LICENSE'),
+];
 
+/** Builds components to UMD bundle. */
+task('build:components', [':build:components:bundle:umd']);
 
-/** [Watch task] Rebuilds (ESM output) whenever ts, scss, or html sources change. */
-task(':watch:components', () => {
-  watch(path.join(COMPONENTS_DIR, '**/*.ts'), ['build:components', triggerLivereload]);
-  watch(path.join(COMPONENTS_DIR, '**/*.scss'), ['build:components', triggerLivereload]);
-  watch(path.join(COMPONENTS_DIR, '**/*.html'), ['build:components', triggerLivereload]);
-});
-
+/** Builds components for Angular Material releases */
+task(':build:components:release', sequenceTask(
+  ':build:components:bundle:umd',
+  ':build:components:bundle:esm',
+  ':build:components:ngc'
+));
 
-/** Builds component typescript only (ESM output). */
-task(':build:components:ts', tsBuildTask(path.join(COMPONENTS_DIR, 'tsconfig-srcs.json')));
+/** Builds components typescript in ES5, ES6 target. For specs Karma needs CJS output. */
+task(':build:components:ts:es5', tsBuildTask(tsconfigPath, { target: ScriptTarget.ES5 }));
+task(':build:components:ts:es6', tsBuildTask(tsconfigPath, { target: ScriptTarget.ES6 }));
```

Comment:
@DevVersion - Worth noting that `ES6` as a ScriptTarget doesn't work for newer versions of TypeScript iirc.

Options are `ES2015, ES2016, ES2017, ES3, ES5, ESNext & Latest`